libraries/eosiolib/tests/test-sdk.cpp

```diff
@@ -59,3 +61,42 @@ TEST_CASE("get_code")
 
    // clsdk/bin/cleos push action getcode get '["eosio"]' -p getcode -j | jq .processed.action_traces[0].return_value_data
 }
+
+template <typename T>
+void convert_container(T& dest, const T& src)
+{
+   dest = src;
+}
+
+void convert_container(eosio::input_stream& dest, const eosio::bytes& src)
+{
+   dest = {src.data.data(), src.data.size()};
+}
+
+TEST_CASE("hex_serialization")
```

Comment:
abieos/src/template_test.cpp tests that all combinations of serialization with and without ABIs round trip.